integration_test.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+set -Eemo pipefail
+# This script is meant to be run inside https://hub.docker.com/_/docker
+# See .github/workflows/ci.yml:18 for how to run this yourself
+
+# Wait for the docker daemon to start up
+for i in {1..30}; do
+    if docker ps; then
+        break
+    else
+        sleep 1
+    fi
+done
+if [ "$i" = '30' ]; then
+    echo "Docker did not start within 30 seconds, aborting"
```

Comment:
For style, can we add a Unicode ellipsis character after `aborting`? Maybe:

```sh
echo 'Docker did not start within 30 seconds. Aborting…'
```

I also prefer single quotes in shell scripts for safety unless double quotes are actually needed (e.g., for interpolating variables).

One more suggestion: print to stderr instead of stdout:

```sh
echo 'Docker did not start within 30 seconds. Aborting…' >&2
```

Reply:
(can we also apply these suggestions to the `echo`s in the rest of this script)